packages/waku/src/router/fs-router.ts

```diff
@@ -157,7 +157,16 @@ export function unstable_fsRouter(
             ...config,
           });
         } else if (pathItems.at(-1)?.startsWith('_part')) {
-          const order = parseInt(pathItems.at(-1)!.split('-')[0]!.substring(5));
+          const order = parseInt(
+            pathItems.at(-1)!.split('-')[0]!.slice('_part'.length),
+          );
+          if (isNaN(order)) {
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN

> It is a more robust version of the original, global [isNaN()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN) function.

```suggestion
          if (Number.isNaN(order)) {
```